test/cases/lifecycle.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+
+let log = [];
+const logger = function(msg) {
+	return function() {
+		// return true for shouldComponentUpdate
+		log.push(msg);
+		return true;
+	};
+};
+
+export const getLogMessages = () => {
+	const returnedLog = log;
+	log = [];
+	return returnedLog;
+};
+
+export class Outer extends React.Component {
+	constructor() {
+		super();
+
+		this.UNSAFE_componentWillMount = logger('outer componentWillMount');
```

Comment:
I intentionally omitted these UNSAFE methods, since they are going to be removed anyway.